### DISCUSSION TITLE
How to identify which snapshot button is attached to which camera?

### DISCUSSION THREAD

**Discussion by supersebbo** [tsightler/ring-mqtt#744] [Support Q&A] [answered]
Hi,

I would like to start using the snapshot request button on my ring cameras.  I have located the 'button' entities but I cannot find a way to work out which button is related to which camera?  They are just generically named 1,2,3.

If there a way to see what this mapping is?  

Thanks!

![image](https://github.com/tsightler/ring-mqtt/assets/29370445/1d616c8c-f2c1-4421-975b-532fa9262e75)
![image](https://github.com/tsightler/ring-mqtt/assets/29370445/20747ce2-0edb-4127-8f6e-545c82092612)

**Comment by supersebbo** [comment]
Ok, so there is a way, if you open up the device page of the camera, then open the more info panel of the Button, then the config modal, you see the entity name.  

Perhaps a feature request to append the camera name to the button name instead of a number?

**Reply by tsightler** [reply]
Is there any chance you haven't updated to a recent HA version from the last couple of months?  Prior to v5.6.0 of ring-mqtt the entity names did include device name, however, as of Home Assistant 2023.8.0, the HA team made changes to the MQTT entities to align with other platforms and started issuing warning for any entities that included the device name as part of the entity name, so ring-mqtt was modified to remove this.

If you are on a recent version of HA (i.e. anything after 2023.8.x) and see this behavior, then that is an HA bug, however I haven't see this.  Out of curiosity, do the other entities that are part of the camera device also show this behavior or do their entity IDs properly include the device name and it's only the snapshot button?  If it's the latter then it might be an HA bug with the button MQTT entity only although again, I haven't seen this.  For example, the generated entity ID for my Front Doorbell camera is `button.front_door_take_snapshot` which is exactly what is expect as it's name in Ring is "Front Door".

Note that you can rename the entity IDs to anything you want, the name you are shoring is just what HA generates automatically based on it's naming rules. The ring-mqtt addon simply sends the device discovery information via MQTT that includes the device id and name, as well as the entity name, and HA then generates an entity ID based on that data.  The auto-generated rule in HA 2023.8.x and later should create an id with device_name+entity_name, but, if it doesn't I can't really control it, but you can change it.

**Reply by supersebbo** [reply]
I am on 2023.10.0 - however I had a backlog of updates to apply to HA and installed the Ring-MQTT update _before_ the HA update, so that may have been the issue.  All other entities are named with the camera name: 
![image](https://github.com/tsightler/ring-mqtt/assets/29370445/f90c1ad4-9e2d-41a3-a4e4-110425e942b0)

**Reply by supersebbo** [reply]
Interestingly there is also a mix of 'front_door_XXXXX' anf 'front_doorbell_XXXXX' which is likley due to me renaming the device at some point... entity naming seems to be a bit of a mess in general in HA.

**Reply by tsightler** [reply, marked as answer]
To be fair to the HA team, the changes they made to MQTT entity naming in 2023.8.0 are a step in the right direction, bringing MQTT entity naming inline with other platforms, I just hated the way they rolled out those changes, issuing dozens/hundreds of scary sounding warnings to users that don't really understand the guts of what is happening.  The rules are pretty clear now though, the entity should be the function name, and entity ID will be device_name+entity_name.  I'm guessing your out-of-order update could have caused this change in your case, although I'm not 100% sure how because entity_id naming didn't really change, only the display name.

Regardless, I don't think there's anything I can do from ring-mqtt side, I send the information HA requests, and it's up to HA to generate the ID based on its rules.
